src/webdrivers/browser_factory.py

```diff
@@ -0,0 +1,43 @@
+from typing import Union
+
+from selenium import webdriver
+from loguru import logger
+
+from config import BROWSER_TYPE_CONFIG
+from src.webdrivers.base_browser import BrowserType
+
+
+class BrowserFactory:
+    """Factory class for creating browser instances"""
+    _browser_type: BrowserType = BROWSER_TYPE_CONFIG
+
+    @classmethod
+    def get_browser_type(cls) -> BrowserType:
+        """Get current browser type"""
+        return cls._browser_type
+
+    @classmethod
+    def set_browser_type(cls, browser_type: BrowserType) -> None:
+        """Set browser type"""
+        if browser_type not in BrowserType.__members__:
+            raise ValueError(f"Unsupported browser type: {browser_type}")
+        cls._browser_type = browser_type
+        logger.debug(f"Browser type set to: {browser_type}")
+
+    @classmethod
+    def get_driver(cls, browser_type: BrowserType) -> Union[webdriver.Chrome, webdriver.Firefox]:
+        """
+        Create and return a WebDriver instance for the specified browser type
+        Args:
+            browser_type: BrowserType enum value
+        Returns:
+            WebDriver instance
+        Raises:
+            RuntimeError: If browser initialization fails
+        """
+        browser_class = browser_type.value
+        if not browser_class:
+            raise ValueError(f"Unsupported browser type: {browser_type}")
+
+        browser = browser_class()
```

Comment:
implement singleton. Don't create multiple instances of browser driver as they are resource heavy.

Reply:
not sure if there is a better way, but if we want to implement singleton in here, we need to maintain a browser class map within the BrowserFactory. this approach could compromise the principle of having a single source of truth.